pkg/murmur3/murmur3.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package murmur3
+
+import (
+	"unsafe"
+)
+
+// Hash128 calculates a 128 bits hash for the given data. It returns different
+// results when running on big-endian and little-endian machines.
+func Hash128(data []byte, seed uint32) (uint64, uint64) {
+	nblocks := len(data) / 16
+
+	h1 := uint64(seed)
+	h2 := uint64(seed)
+
+	c1 := uint64(0x87c37b91114253d5)
```

Comment:
Should this be a constant with a descriptive name and/or comment?

Reply:
I wanted to keep the original vars naming as in the reference cpp implementation. I will make the `c{1,2}` as a constant though.

Reply:
I was referring to the numeric hex value... I still have no idea whether this is a magic number of sorts or just a randomly chosen number?

Reply:
The consts were chosen by the original author of the algorithm.